msal/src/main/java/com/microsoft/identity/client/TokenParameters.java

```diff
@@ -202,6 +204,18 @@ public B callback(AuthenticationCallback callback) {
             return self();
         }
 
+        public B withResource(final String resource) {
+            if (!StringUtil.isEmpty(resource)) {
+                mScopes = new ArrayList<String>() {{
```

Comment:
We should check if `null != mScopes` and, if it is, throw an `Exception` (perhaps `IllegalStateException` or `IllegalArgumentException`) [since scopes and resources cannot be combined in a single request](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#the-default-scope).

The same logic would need to be added to `withScopes(List)` to ensure the behavior is the same whether `withScopes()` or `withResource()` is called first

Reply:
Thank you, Brian! Addressed.